pulsar/producer_impl.go

```diff
@@ -319,12 +315,26 @@ func (p *producer) getPartition(msg *ProducerMessage) Producer {
 	return producers[partition]
 }
 
+func (p *producer) getProducers() []Producer {
+	if producers := p.producers.Load(); producers != nil {
+		return producers.([]Producer)
+	}
+	return []Producer{}
```

Comment:
It's better to panic here. Otherwise, method call like `getProducer(partition)` could also panic but it's hard to see if it's caused by the out of range of `partition`.

Reply:
Yes. Add a panic check in the `getProducer(partition)`.

This method is also used at https://github.com/apache/pulsar-client-go/blob/7940ff88dd91b2af3b721b0e4c0cf4910bafdc53/pulsar/producer_impl.go#L194 so it's expected to return the empty producers slice during the init.